### PR TITLE
Frontend main layout

### PR DESCRIPTION
**Pull request by Nat3005**
Main layout consisting of top navigation and sidebar for responsive view, full routing.

### CODE REVIEW

File: frontend/src/pages/calculator/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  return (
+    <div style={{display: 'flex', justifyContent:'center', alignItems: 'center', height: '90vh'}}>
+      <h1>
+        Placeholder for calculator
+      </h1>
+  </div>
+);
+};
+
+export default Calculator
```
Comment (kkuriata) [inline frontend/src/pages/calculator/index.js:14]: New line

File: frontend/src/pages/howToCalculateInstallments/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  return (
+    <div style={{display: 'flex', justifyContent:'center', alignItems: 'center', height: '90vh'}}>
+        <h1>
+          Placeholder for how we calculate installments
+        </h1>
+    </div>
+  );
+};
+
+export default HowToCalculateInstallments
```
Comment (kkuriata) [inline frontend/src/pages/howToCalculateInstallments/index.js:13]: New line

File: frontend/src/pages/howToCalculateInstallments/index.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+
+const HowToCalculateInstallments = () => {
+  return (
+    <div style={{display: 'flex', justifyContent:'center', alignItems: 'center', height: '90vh'}}>
```
Comment (kkuriata) [inline frontend/src/pages/howToCalculateInstallments/index.js:5]: Styled component

File: frontend/src/pages/index/index.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import './style.css'
+
+
+const Home = () => {
+  return (
+    <div style={{display: 'flex', justifyContent:'center', alignItems: 'center', height: '90vh'}}>
```
Comment (kkuriata) [inline frontend/src/pages/index/index.js:7]: Styled component

File: frontend/src/pages/index/index.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+      <h1>
+        Placeholder for homepage
+      </h1>
+  </div>
+);
+};
+
+export default Home;
```
Comment (kkuriata) [inline frontend/src/pages/index/index.js:15]: New line

File: frontend/src/pages/usefulTerms/usefulTerms.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+
+const UsefulTerms = () => {
+  return (
+    <div style={{display: 'flex', justifyContent:'center', alignItems: 'center', height: '90vh'}}>
```
Comment (kkuriata) [inline frontend/src/pages/usefulTerms/usefulTerms.js:5]: Styled component

File: frontend/src/pages/usefulTerms/usefulTerms.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+        <h1>
+            Placeholder for useful terms
+        </h1>
+    </div>
+  );
+};
+
+export default UsefulTerms
```
Comment (kkuriata) [inline frontend/src/pages/usefulTerms/usefulTerms.js:13]: Empty line and semicolon

File: package.json [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (kkuriata) [inline package.json:1]: Why do you need `package.json` (and lock) here? I guess someone installed this dependency by mistake

**Review by kkuriata — CHANGES_REQUESTED** [review]
Some of the comments are the same (just to not miss anything). Anyways, there is some problem w with linter that does not check formatting and styling. Try to fix that. If you have any problems, write on Slack and I will try to have a look at it

File: frontend/src/App.css [hunk 9]
```diff
@@ -1,38 +1,8 @@
...
-  }
-}
+@import url('https://fonts.googleapis.com/css2?family=Comic+Neue:wght@700&family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&display=swap');
+
+*{
+  font-family: 'Roboto', sans-serif;
+  box-sizing: border-box;
+  margin: 0;
+  padding: 0;
+}
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/src/App.css:8]: done

File: frontend/src/App.js [hunk 10]
```diff
@@ -1,25 +1,31 @@
...
+import Home from './pages/index';
+import Calculator from './pages/calculator';
+import HowToCalculateInstallments from './pages/howToCalculateInstallments';
+import UsefulTerms from './pages/usefulTerms/usefulTerms';
+import CalculatedResults from './pages/calculatedResults';
+import Sidebar from './components/sidebar';
+import React, {useState} from 'react';
 function App() {
+  const[isOpen, setIsOpen] = useState(false)
+  const toggle = ()=>{
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/src/App.js]: done

File: frontend/src/App.js [hunk 11]
```diff
@@ -1,25 +1,31 @@
...
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <BrowserRouter>
+      <Sidebar isOpen={isOpen} toggle={toggle}/>
+      <Navbar toggle={toggle}/>
+    <Routes>
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/src/App.js]: done

File: frontend/src/components/navbar/NavbarElements.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
+import styled from 'styled-components'
+import { NavLink as Link } from 'react-router-dom'
+import {FaBars} from 'react-icons/fa'
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/src/components/navbar/NavbarElements.js]: done

File: frontend/src/components/navbar/NavbarElements.js [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+export const Nav = styled.nav`
+    background: linear-gradient(88.54deg, #271F7F -0.21%, #5E60CE 60.5%, #00FFD4 112.08%);
+    height: 80px;
+    display: flex;
+    justify-content: space-around;
+    padding: 0.5rem calc((100vw - 1000px) /2);
+    z-index: 10000;
+
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/src/components/navbar/NavbarElements.js]: done

File: frontend/src/pages/calculatedResults/index.js [hunk 14]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import "./style.css";
+
+const CalculatedResults = () => {
+
+  return (
+      <div style={{display: 'flex', justifyContent:'center', alignItems: 'center', height: '90vh'}}>
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/src/pages/calculatedResults/index.js]: done

File: frontend/src/components/sidebar/index.js [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+            <SidebarLink to="/jak-obliczamy-rate" onClick={toggle}>Jak obliczamy ratę?</SidebarLink>
+            <SidebarLink to="/przydatne-pojecia" onClick={toggle}>Przydatne pojęcia</SidebarLink>
+        </SidebarMenu>
+    </SidebarWrapper>
+
+    </SidebarContainer>
+  )
+}
+
+export default Sidebar;
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/src/components/sidebar/index.js]: done

File: frontend/src/components/sidebar/SidebarElements.js [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+    color: #271F7F;
+    cursor: pointer;
+    padding: 1rem;
+
+    &:hover{
+        color: #5E60CE;
+        transition: 0.2s ease-in-out;
+    }
+
+`
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/src/components/sidebar/SidebarElements.js]: done

File: frontend/src/components/sidebar/SidebarElements.js [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+    width: 100%;
+    height: 100%;
+    display: grid;
+    align-items: center;
+    top: 0;
+    right: 0;
+    transition: 0.5s ease-in-out;
+    opacity: ${({isOpen}) =>(isOpen ? '100%' : '0')};
+    right: ${({isOpen})=>(isOpen ? '0' : '-100%')};
+
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/src/components/sidebar/SidebarElements.js]: done

File: frontend/src/components/navbar/index.js [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+                <HamburgerIcon />
+            </Icon>
+            <NavMenu>
+                <NavBtn>
+                    <NavBtnLink to="/kalkulator">Kalkulator</NavBtnLink>
+                </NavBtn>
+                <NavLink to="/jak-obliczamy-rate" >Jak obliczamy ratę?</NavLink>
+                <NavLink to="/przydatne-pojecia" >Przydatne pojecia</NavLink>
+            </NavMenu>
+
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/src/components/navbar/index.js:13]: done

File: frontend/src/components/navbar/NavbarElements.js [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+export const NavLink = styled(Link)`
+    color: #fff;
+    align-items: center;
+    text-decoration: none;
+    padding:  1.5rem;
+    height: 100%;
+    cursor: pointer;
+    display: flex;
+
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/src/components/navbar/NavbarElements.js]: done

File: frontend/src/pages/calculator/index.js [hunk 20]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import "./style.css";
+
+const Calculator = () => {
+  return (
+    <div style={{display: 'flex', justifyContent:'center', alignItems: 'center', height: '90vh'}}>
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/src/pages/calculator/index.js]: done

File: frontend/src/pages/calculator/index.js (hunk 1 above)
Comment (Nat3005) [reply frontend/src/pages/calculator/index.js]: done

File: frontend/src/pages/howToCalculateInstallments/index.js (hunk 2 above)
Comment (Nat3005) [reply frontend/src/pages/howToCalculateInstallments/index.js]: done

File: frontend/src/pages/howToCalculateInstallments/index.js (hunk 3 above)
Comment (Nat3005) [reply frontend/src/pages/howToCalculateInstallments/index.js]: done

File: frontend/src/pages/index/index.js (hunk 4 above)
Comment (Nat3005) [reply frontend/src/pages/index/index.js]: done

File: frontend/src/pages/index/index.js (hunk 5 above)
Comment (Nat3005) [reply frontend/src/pages/index/index.js]: done

File: frontend/src/pages/usefulTerms/usefulTerms.js (hunk 6 above)
Comment (Nat3005) [reply frontend/src/pages/usefulTerms/usefulTerms.js]: done

File: frontend/src/pages/usefulTerms/usefulTerms.js (hunk 7 above)
Comment (Nat3005) [reply frontend/src/pages/usefulTerms/usefulTerms.js]: done

File: package.json (hunk 8 above)
Comment (Nat3005) [reply package.json]: removed

File: frontend/.eslintrc.js [hunk 21]
```diff
@@ -19,5 +20,6 @@ module.exports = {
         "react"
     ],
     "rules": {
+        "react/prop-types": "off"
```
[the remark this answers is not in the document]
Comment (Nat3005) [reply frontend/.eslintrc.js]: we couldn't find the solution yesterday but we were able to apply changes today that don't require this rule

**Review by mwidera — APPROVED** [review]
Minor changes: missing new line at EOL, thus I'm to approve

Before merge please address @kkuriata comments

**Comment by Nat3005** [comment]
.css changed to .js to include styled components throughout the entire project, so all files have new line at the end

**Review by mwidera — APPROVED** [review]

**Review by kkuriata — APPROVED** [review]

**Pull request merged**